File: android/app/src/main/kotlin/pt/ualg/miaugenda/data/remote/WeekAssignmentApi.kt
package pt.ualg.miaugenda.data.remote

import pt.ualg.miaugenda.data.model.CreateWeekAssignmentRequest
import pt.ualg.miaugenda.data.model.WeekAssignment
import retrofit2.Response
import retrofit2.http.*

interface WeekAssignmentApi {
    @GET("/week-assignments")
    suspend fun getWeekAssignments(@Query("week") week: String): Response<List<WeekAssignment>>

    @POST("/week-assignments")
    suspend fun createWeekAssignment(@Body request: CreateWeekAssignmentRequest): Response<WeekAssignment>

    @DELETE("/week-assignments/{id}")
    suspend fun deleteWeekAssignment(@Path("id") id: Int): Response<Unit>
}

File: desktop/src/api/weekAssignmentApi.ts
import { apiClient } from "./apiClient";
import type {
  CreateWeekAssignmentRequest,
  WeekAssignment
} from "../models/weekAssignment";

export async function getWeekAssignments(
  week: string
): Promise<WeekAssignment[]> {
  const response = await apiClient.get<WeekAssignment[]>("/week-assignments", {
    params: {
      week
    }
  });

  return response.data;
}

export async function createWeekAssignment(
  request: CreateWeekAssignmentRequest
): Promise<WeekAssignment> {
  const response = await apiClient.post<WeekAssignment>(
    "/week-assignments",
    request
  );

  return response.data;
}

export async function deleteWeekAssignment(id: number): Promise<void> {
  await apiClient.delete(`/week-assignments/${id}`);
}